router/core/plan_generator.go

```diff
@@ -119,6 +119,8 @@ func NewPlanGenerator(configFilePath string, logger *zap.Logger) (*PlanGenerator
 		pg.planConfiguration.Logger = log.NewZapLogger(logger, log.DebugLevel)
 	}
 
+	pg.planConfiguration.MaxDataSourceCollectorsConcurrency = maxDataSourceCollectorsConcurrency
```

Comment:
Why is this set via a flag instead of whatever the rest of the config comes from? `pg.LoadConfiguration`?